src/sentry/ownership/grammar.py

```diff
@@ -109,29 +109,47 @@ def test(self, data):
         elif self.type.startswith("tags."):
             return self.test_tag(data)
         elif self.type == CODEOWNERS:
-            return self.test_codeowners(data)
+            return self.test_frames(
+                data,
+                ["filename", "abs_path"],
+                # Codeowners has a slightly different syntax compared to issue owners
+                # As such we need to match it using gitignore logic.
+                # See syntax documentation here:
+                # https://docs.github.com/en/github/creating-cloning-and-archiving-repositories/creating-a-repository-on-github/about-code-owners
+                lambda val, pattern: _path_to_regex(pattern).search(val),
+            )
         return False
 
-    def test_url(self, data):
+    def test_url(self, data: Union[Mapping[str, Any], Sequence[Any]]) -> bool:
```

Comment:
Since this data type is reused a lot it might be worth declaring it separately as it's own type.

I wonder if we have a type for this somewhere, since it should be the value of `event.data`